autoprotocol/protocol.py

```diff
@@ -1154,14 +1154,14 @@ def format_source_well(well: Well, num_chips: int) -> Tuple[Well, int]:
                     )
                 destination: List[WellGroup] = [WellGroup(d) for d in destination]
 
-        for (sw, number_of_chips), destinations in zip(source, destination):
-            # Check that the number of destinations match the number of chips specified for each
-            # source destination mapping
-            if len(destinations) != number_of_chips:
-                raise ValueError(
-                    f"The number of destinations {len(destinations)} does not match the number of "
-                    f"chips {number_of_chips} specified for the given source {(sw, number_of_chips)}"
-                )
+                for (sw, number_of_chips), destinations in zip(source, destination):
+                    # Check that the number of destinations match the number of chips specified for each
+                    # source destination mapping
+                    if len(destinations) != number_of_chips:
+                        raise ValueError(
+                            f"The number of destinations {len(destinations)} does not match the number of "
+                            f"chips {number_of_chips} specified for the given source {(sw, number_of_chips)}"
+                        )
```

Comment:
we dont necessarily need the number of chips to match the number of destinations right?

we should be able to specify 

`sources = [(tube1.well(0), chips=4), (tube2.well(0), chips=1)]` with
 `destinations =[WellGroup(plate.A1), WellGroup(plate.A1)]`

but in this case, `len(src) = len(dest)`, but `len(destination[0]) != len(sources[0][1])`, which would trigger the above error message

Reply:
if i understand, we should be able to just remove this check completely since we dont really ever need the number of chips to match anything. we just have to make sure we dont have more chips than the total available - but you already check for that above